File: dotnet-desktop-guide/wpf/app-development/snippets/how-to-use-host-builder/csharp/GreetingService.cs
using Microsoft.Extensions.Configuration;

namespace HostBuilderApp;

// <GreetingService>
public class GreetingService(IConfiguration configuration) : IGreetingService
{
    public string GetGreeting() =>
        configuration.GetValue<string>("GreetingMessage")
        ?? "Hello from the Generic Host!";
}
// </GreetingService>

File: dotnet-desktop-guide/wpf/app-development/snippets/how-to-use-host-builder/vb/GreetingService.vb
Imports Microsoft.Extensions.Configuration

' <GreetingService>
Public Class GreetingService
    Implements IGreetingService

    Private ReadOnly _configuration As IConfiguration

    Public Sub New(configuration As IConfiguration)
        _configuration = configuration
    End Sub

    Public Function GetGreeting() As String Implements IGreetingService.GetGreeting
        Dim message As String = _configuration.GetValue(Of String)("GreetingMessage")

        If String.IsNullOrEmpty(message) Then
            Return "Hello from the Generic Host!"
        End If

        Return message
    End Function

End Class
' </GreetingService>
